services/main-frontend/src/pages/signup.tsx

```diff
@@ -0,0 +1,280 @@
+      font-weight: 600;
+      text-decoration: none;
+    }
+  }
+
+  fieldset {
+    border: none;
+    margin: 0;
+    padding: 0;
+  }
+
+  input {
+    background: #fcfcfc;
+    border-width: 1.6px;
+    border-style: solid;
+    border-color: #bec3c7;
+    padding: 10px;
+    transition: ease-in-out, width 0.35s ease-in-out;
+    outline: none;
+    min-width: 20px;
+    width: 100%;
+    display: inline-block;
+    font-size: 18px;
+    margin-bottom: 14px;
+
+    &:focus,
+    &:active {
+      border-color: #55b3f5;
+    }
+
+    @media (max-width: 767.98px) {
+      padding: 10px 8px;
+    }
+  }
+
+  label {
+    display: inline-block;
+    font-size: 17px;
+    margin-bottom: 5px;
+  }
+
+  input[type="submit"] {
+    height: 60px;
+    margin-top: 24px;
+    background: #46749b;
+    color: #fff;
+    font-weight: bold;
+    font-size: 22px;
+    padding: 15px 10px;
+    line-height: 1.2;
+    font-family: ${headingFont} !important;
+    justify-content: center;
+    align-items: center;
+    border: none;
+
+    &:hover {
+      background: #215887;
+    }
+  }
+
+  input[type="submit"]:disabled {
+    background: #ebedee;
+    color: #989ca3;
+  }
+
+  .signin-link {
+    display: block;
+    text-align: center;
+    margin: 0 auto;
+
+    a {
+      text-decoration: none;
+      font-size: 20px;
+      color: ${baseTheme.colors.grey[700]};
+
+      &:hover {
+        color: ${baseTheme.colors.blue[700]};
+      }
+    }
+  }
+`
+
+const CreateAccountForm: React.FC = () => {
+  // eslint-disable-next-line i18next/no-literal-string
+  const { register, formState, watch, reset, handleSubmit } = useForm({ mode: "onChange" })
+  const loginStateContext = useContext(LoginStateContext)
+  const router = useRouter()
+  const { errors, isValid, isSubmitSuccessful, isSubmitting } = formState
+
+  const [submitError, setSubmitError] = useState(false)
+
+  const { t, i18n } = useTranslation()
+
+  // eslint-disable-next-line i18next/no-literal-string
+  const password = watch("password")
+
+  useEffect(() => {
+    if (loginStateContext.signedIn) {
+      router.push("/")
+    }
+    reset({
+      first_name: "",
+      last_name: "",
+      email: "",
+      password: "",
+      password_confirmation: "",
+    })
+  }, [isSubmitSuccessful, loginStateContext.signedIn, reset, router])
+
+  return (
+    <Wrapper>
+      <h2>{t("create-new-account")}</h2>
+      <span className="description">{t("sign-up-with-mooc-subtitle")}</span>
+      <form
+        onSubmit={handleSubmit(async (data, event) => {
+          event?.preventDefault()
+          const { first_name, last_name, email, password, password_confirmation } = data
+          try {
+            await createUser({
+              email: email,
+              first_name: first_name,
+              last_name: last_name,
+              language: i18n.language,
+              password: password,
+              password_confirmation: password_confirmation,
+            })
```

Comment:
Do the reset here

Reply:
I forgot to put the reset in an if-block inside the useEffect. I've rectified that now, because according to the useForm docs - it is better to put reset() function in useEffect because handleSubmit handle an async function.